packages/wrangler/src/__tests__/init.test.ts

```diff
@@ -1900,6 +1902,184 @@ describe("init", () => {
+				mockResponse: mockDashboardScript,
+			});
+			mockConfirm(
+				{
+					text: "Would you like to use git to manage this Worker?",
+					result: false,
+				},
+				{
+					text: "Would you like to use TypeScript?",
+					result: true,
+				},
+				{
+					text: "No package.json found. Would you like to create one?",
+					result: true,
+				},
+				{
+					text: "Would you like to install the type definitions for Workers into your package.json?",
+					result: true,
+				}
+			);
+
+			await runWrangler("init  --from-dash isolinear-optical-chip");
+
+			checkFiles({
+				items: {
+					"isolinear-optical-chip/src/index.js": false,
+					"isolinear-optical-chip/src/index.ts": {
+						contents: mockDashboardScript,
+					},
+					"isolinear-optical-chip/package.json": {
+						contents: expect.objectContaining({
+							name: "isolinear-optical-chip",
+						}),
+					},
+					"isolinear-optical-chip/tsconfig.json": true,
+					"isolinear-optical-chip/wrangler.toml": wranglerToml({
+						...MINIMAL_WRANGLER_TOML,
+						name: "isolinear-optical-chip",
+					}),
+				},
+			});
+		});
+
+		it("should download source script from dashboard as plain JavaScript", async () => {
+			setMockFetchDashScript({
+				accountId: "LCARS",
+				fromDashScriptName: "isolinear-optical-chip",
+				mockResponse: mockDashboardScript,
+			});
+			mockConfirm(
+				{
+					text: "Would you like to use git to manage this Worker?",
+					result: false,
+				},
+				{
+					text: "Would you like to use TypeScript?",
+					result: false,
+				},
+				{
+					text: "No package.json found. Would you like to create one?",
+					result: true,
+				},
+				{
+					text: "Would you like to install the type definitions for Workers into your package.json?",
+					result: true,
+				}
+			);
+
+			await runWrangler("init  --from-dash isolinear-optical-chip");
+
+			checkFiles({
+				items: {
+					"isolinear-optical-chip/src/index.js": {
+						contents: mockDashboardScript,
+					},
+					"isolinear-optical-chip/src/index.ts": false,
+					"isolinear-optical-chip/package.json": {
+						contents: expect.objectContaining({
+							name: "isolinear-optical-chip",
+						}),
+					},
+					"isolinear-optical-chip/tsconfig.json": false,
+					"isolinear-optical-chip/wrangler.toml": wranglerToml({
+						...MINIMAL_WRANGLER_TOML,
+						name: "isolinear-optical-chip",
+					}),
+				},
+			});
+		});
+
+		it("should not continue if no worker name is provided", async () => {
+			await expect(
+				runWrangler("init  --from-dash")
+			).rejects.toMatchInlineSnapshot(
+				`[YError: Not enough arguments following: from-dash]`
+			);
+			checkFiles({
+				items: {
+					"isolinear-optical-chip/src/index.js": false,
+					"isolinear-optical-chip/src/index.ts": false,
+					"isolinear-optical-chip/package.json": false,
+					"isolinear-optical-chip/tsconfig.json": false,
+					"isolinear-optical-chip/wrangler.toml": false,
+				},
+			});
+		});
+	});
```

Comment:
Can we also provide a test case for when they specify a worker name that doesn't exist?